src/transformers/generation_utils.py

```diff
@@ -874,7 +892,15 @@ def generate(
             warnings.warn(
                 "Both `max_length` and `max_new_tokens` have been set but they serve the same purpose.", UserWarning
             )
-
+        if stopping_criteria is not None:
+            for stopping_criterion in stopping_criteria:
+                if isinstance(stopping_criterion, MaxLengthCriteria):
+                    if max_length is not None:
+                        warnings.warn(
+                            "A stopping criteria of type `MaxLengthCriteria` as well as `max_length` was passed to `generate`. The `MaxLengthCriteria` will be used.",
```

Comment:
That's not necessarily true.

On line 889, we can defined `max_length` to `self.config.max_length` even if it's not user defined.
At least we override here which makes behavior what a user would expect IMO (respecting it's own defined things)

Reply:
Oh that's right, I missed that one. I think the `stopping_criteria` should be moved before that. Otherwise you will always get a warning if neither `max_length is None and max_new_tokens is None` which would be the standard case when using a custom `MaxLengthCriteria`. What do you think?